### PR TITLE
Added Lab 9

### PR DESCRIPTION
**Pull request by borjasotomayor**
Lab 9 files are here: https://mit.cs.uchicago.edu/cmsc22000-spr-18/cmsc22000-spr-18/tree/lab9/lab9

### CODE REVIEW

File: website/content/labs/lab9.md [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+2. A developer working on your program (yourself): while developing new features, you likely want to see a rough trace of the structure of your program, without overwhelming yourself with output. There isn't really a clear brightline here on what belongs at the `WARN` through `DEBUG` log levels - use your best judgement.
+3. A developer debugging your program (yourself): note that debugging is a distinct case from normal development. When debugging, it's helpful to get as much output as possible, to be able to inspect the output more clearly. The `TRACE` log level should thus contain the pretty-printed contents of all important data structures and variables whenever they are modified. Take into account that we already provide a function that prett-prints a tic-tac-toe board, but which does not use a logging library to do so. You are welcome to modify this function so that it uses log functions.
```
Comment (lydiafilipe) [inline website/content/labs/lab9.md]: typo- prett-prints

File: website/content/labs/lab9.md [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+* `RECURSIVE` specifies that we want Doxygen to go into subdirectories when searching for files to generate documentation from.
+
+This is only the tip of the iceberg, and there are a ton of other ways you can specify how the file is generated. A list of all the tags can be found here.
+https://www.stack.nl/~dimitri/doxygen/manual/config.html 
+
+Go ahead and re-run doxygen:
+
+	$ doxygen
+And re-load the `index.html` file inside the `html` directory. You will now see a "Classes" tab you can click on. This will show you the structs (or classes, in object oriented parlance) in your code. You should see two types: `point_t` and `polygon_t`. If you click on polygon_t you'll see that page contains abundant documentation about the `polygon_t` struct, its fields, and the functions associated witht it. It even shows a helpful diagram showing how `polygon_t` relates to our other types; in particular, it shows that `polygon_t` has a field called `points` that relies on the `point_t` type. We can even click on `point_t` type to see the documentation for that type.
```
Comment (lydiafilipe) [inline website/content/labs/lab9.md]: typo- "witht"

File: website/content/labs/lab9.md (hunk 1 above)
Comment (borjasotomayor) [reply website/content/labs/lab9.md]: Fixed in 94c0bed

File: website/content/labs/lab9.md (hunk 2 above)
Comment (borjasotomayor) [reply website/content/labs/lab9.md]: Fixed in 94c0bed

**Pull request merged**